doc/tutorials/rtvm_performance_tips.rst

```diff
@@ -0,0 +1,213 @@
+.. _rt_perf_tips_rtvm:
+
+ACRN Real-Time VM Performance Tips
+##################################
+
+Background
+**********
+
+The ACRN real-time VM (RTVM) is a special type of ACRN post-launched VM. In
+order to achieve bare metal-like RT performance, a set of constraints and
+technologies are applied to the RTVM compared to the ACRN standard VM. With
+these additional constraints and technologies, RT tasks can run on the RTVM
+without a VM-exit, which is a key virtualization overhead issue.
+
+In addition to the VM-exit, interference from neighbor VMs, such as Service
+VMs, Human-Machine-Interface (HMI) VMs, or other RT VMs may affect the
+execution of real-time tasks on a certain RTVM. Other technologies are
+applied to isolate noise from the neighbor VMs.
+
+Here is the list of key technologies applied to enable the bare metal-like
+RT performance:
+
+- LAPIC passthrough with core partitioning.
+- PCIe Device Passthrough: Only MSI interrupt-capable PCI devices will be
+  supported for the RTVM.
+- Enable CAT (Cache Allocation Technology)-based cache isolation: RTVM uses
+  a dedicated CLOS (Class of Service). While others may share CLOS, the GPU
+  uses a CLOS that will not overlap with the RTVM CLOS.
+- PMD virtio: Both virtio BE and FE work in polling mode so that the
+  interrupts or notification between the Service VM and RTVM are not needed.
+  The RTVM guest memory is hidden from the Service VM except for the virtio
+  queue memory which is all that the Service VM can access.
+
+This document list tips that are summarized from issues encountered and
+resolved during real-time development and performance tuning.
+
+Mandatory options for an RTVM
+*****************************
+
+An RTVM is a post-launched VM with LAPIC passthrough. To launch an ACRN
+RTVM, take note of the following options:
+
+**Tip 1:** Apply the acrn-dm option "--lapic_pt" and make the guest RTVM
+operate under the LAPIC X2APIC mode to enable the LAPIC passthrough.
+
+The LAPIC passthrough feature of ACRN is configured via the "--lapic_pt"
+option, but the feature is actually enabled when LAPIC is switched to X2APIC
+mode. So, both conditions should be met to enable an RTVM. The "--rtvm"
+option will be automatically attached once "--lapic_pt" is applied.
+
+**Tip 2:** If necessary, use virtio polling mode to prevent the frontend of

```

Comment:
I think the idea is that it's the frontend part of the virtio driver (i.e. the part running in the User VM) that may trigger a `VM-Exit` if polling mode is **not** used. Can we find a better formulation for this here?